### PR TITLE
Lint literal suffixes not separated by underscores (idea also from #703)

### PR DESCRIPTION
**Pull request by birkenfeld**
This changes much the same code as the #703 fix, so the branch is built on that.

### CODE REVIEW

**Comment by birkenfeld** [comment]
Note that this new lint is Allow, because it would otherwise cause LOTs of warnings.

Even for codebases that do `deny(clippy_pedantic)` it might be too much.

File: clippy_lints/src/misc_early.rs [hunk 1]
```diff
@@ -174,7 +206,60 @@ impl EarlyLintPass for MiscEarly {
...
                               "`--x` could be misinterpreted as pre-decrement by C programmers, is usually a no-op");
-    }
+                }
+            }
+            ExprKind::Lit(ref lit) => {
+                if_let_chain! {[
+                    let LitKind::Int(..) = lit.node,
+                    let Some(src) = snippet_opt(cx, lit.span),
+                    let Some(firstch) = src.chars().next(),
+                    char::to_digit(firstch, 10).is_some()
```
Comment (Manishearth) [inline clippy_lints/src/misc_early.rs]: Oh, never mind, macros

File: clippy_lints/src/misc_early.rs [hunk 2]
```diff
@@ -174,7 +206,60 @@ impl EarlyLintPass for MiscEarly {
...
+                ], {
+                    let mut prev = '\0';
+                    for ch in src.chars() {
+                        if ch == 'i' || ch == 'u' {
```
Comment (Manishearth) [inline clippy_lints/src/misc_early.rs]: Also f?

Comment (Manishearth) [inline clippy_lints/src/misc_early.rs]: Er, never mind, you handle those separately.

Not yet fully awake I guess :p

**Pull request merged**